python/tests/test_table.py

```diff
@@ -144,8 +146,15 @@ def test_add_pydantic_model(db):
     class TestModel(LanceModel):
         vector: Vector(16)
         li: List[int]
-
-    data = TestModel(vector=list(range(16)), li=[1, 2, 3])
+        dt: datetime
+        dt_with_tz: datetime = Field(tz="Asia/Kolkata")
+
+    data = TestModel(
+        vector=list(range(16)),
+        li=[1, 2, 3],
+        dt=datetime.now(),
+        dt_with_tz=datetime.now(tz=pytz.timezone("Asia/Kolkata")),
```

Comment:
do we want to check whether tz is appropriately serialized

Reply:
If you put in an invalid timezone pytz will complain. If you put in a tz that doesn't match the schema, then the current behavior is that it'll just get coerced into the timezone that matches the schema.
I was thinking we should ask the user to explicitly add validation to the pydantic model, rather than spreading the validation logic across a bunch of different places. That isn't ideal because if you're not careful you can certainly end up with bad data. But putting this adhoc validator in LanceDB also doesn't feel right 🤷‍♂️